.github/workflows/test-tidy.yml

```diff
@@ -0,0 +1,52 @@
+name: Go mod tidy check
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "**.go"
+  pull_request:
+    paths:
+      - "**.go"
+
+# Abort runs of *this* workflow, if a new commit with the same ref is pushed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  gotidycheck:
+    name: Go mod tidy check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout on PR
+        if: github.event_name == 'pull_request'
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Checkout on Push
+        if: github.event_name == 'push'
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - name: Install Dependencies
+        run: sudo apt-get update && sudo apt-get -y install libcryptsetup-dev libvirt-dev
+
```

Comment:
```suggestion
```

I think you actually don't need this since you are not building anything?

Reply:
If I remember correctly I tried it without and it didn't work. This is also needed for govulncheck and lint action.

Reply:
Running go lint & go mod tidy in Docker without libvirt-dev installed, I don't see any issues.

Not sure about the linter, but govulncheck does static code checking on binaries and thus requires these packages to build them (it's also OS/arch dependent).